frontend/src/scenes/feature-flags/FeatureFlag.tsx

```diff
@@ -523,111 +467,128 @@ export function FeatureFlag(): JSX.Element {
                         </div>
                     )}
 
-                    <h3 className="l3">Release condition groups ({featureFlag.filters.groups.length})</h3>
-                    <div className="text-muted mb">
-                        Specify which users or groups of users to which you want to release this flag.
+                    <div className="feature-flag-form-row">
+                        <div>
+                            <h3 className="l3">Release conditions</h3>
+                            <div className="text-muted mb">
+                                Specify the {aggregationTargetName} to which you want to release this flag. Note that
+                                condition sets are rolled out independently of each other.
+                            </div>
+                        </div>
+                        {showGroupsOptions && (
+                            <div className="centered">
+                                Match by
+                                <Select
+                                    value={
+                                        featureFlag.filters.aggregation_group_type_index != null
+                                            ? featureFlag.filters.aggregation_group_type_index
```

Comment:
non-blocking

just curious, is it possible to use a descriptive string literal for group types instead of an integer or is that not possible?

Reply:
This would result in more code. What does that solve for given it's only visible in redux state/html?

Reply:
Given the case that I don't know what `featureFlag.filters.aggregation_group_type_index` is, I would look at the typing which doesn't reveal much. I can look at `groupTypes` and see what it relates to I suppose. Anyway, non-blocking.